conan/tools/cmake/toolchain/blocks.py

```diff
@@ -730,3 +730,37 @@ def context(self):
                 "cmake_system_name": system_name,
                 "cmake_system_version": system_version,
                 "cmake_system_processor": system_processor}
+
+
+class OutputDirsBlock(Block):
+
+    @property
+    def template(self):
+        if not self._conanfile.package_folder:
+            return ""
+
+        return textwrap.dedent("""
+           set("CMAKE_INSTALL_PREFIX" "{{package_folder}}")
+           set("CMAKE_INSTALL_BINDIR" "{{default_bin}}")
+           set("CMAKE_INSTALL_SBINDIR" "{{default_bin}}")
+           set("CMAKE_INSTALL_LIBEXECDIR" "{{default_bin}}")
+           set("CMAKE_INSTALL_LIBDIR" "{{default_lib}}")
+           set("CMAKE_INSTALL_INCLUDEDIR" "{{default_include}}")
+           set("CMAKE_INSTALL_OLDINCLUDEDIR" "{{default_include}}")
+           set("CMAKE_INSTALL_DATAROOTDIR"] "{{default_res}}")
+        """)
+
+    def _get_cpp_info_value(self, name):
+        # Why not taking cpp.build? because this variables are used by the "cmake install"
+        # that correspond to the package folder (even if the root is the build directory)
+        elements = getattr(self._conanfile.cpp.package, name)
+        return elements[0] if elements else None
```

Comment:
Would it make sense to have a more specific function on ConanFile class side? Something like:

```python
class ConanFile(object):
    # . . .
    @property
    def default_configure_install_dirs(self):
        # Why not taking cpp.build? because this variables are used by the "cmake install"
        # that correspond to the package folder (even if the root is the build directory)
        p = self.cpp.package
        return {
            "includedirs": p.includedirs[0] if len(p.includedirs) else None,
            "libdirs": p.libdirs[0] if len(p.libdirs) else None,
            "bindirs": p.bindirs[0] if len(p.bindirs) else None,
            "resdirs": p.resdirs[0] if len(p.resdirs) else None,
            "builddirs": p.builddirs[0] if len(p.builddirs) else None,
            "frameworkdirs": p.frameworkdirs[0] if len(p.frameworkdirs) else None
        }
```

Reply:
Not in ``ConanFile`` public interface, that is for sure, maybe in ``NewCppInfo`` (the indicator that it could belong there is that almost everything in that logic is ``p.xxxx`` with ``p`` being an object of type ``NewCppInfo``.

Probably not worth extracting yet until this is clear and more stable.